catalog-rest-service/src/main/java/org/openmetadata/catalog/kafka/KafkaEventPublisher.java

```diff
@@ -0,0 +1,82 @@
+package org.openmetadata.catalog.kafka;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.ArrayList;
+import java.util.Properties;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.config.SslConfigs;
+import org.openmetadata.catalog.events.AbstractEventPublisher;
+import org.openmetadata.catalog.resources.events.EventResource;
+import org.openmetadata.catalog.type.ChangeEvent;
+import org.openmetadata.catalog.util.JsonUtils;
+
+@Slf4j
+public class KafkaEventPublisher extends AbstractEventPublisher {
+
+  protected final KafkaEventConfiguration kafkaEventConfiguration;
+  private static KafkaProducer<String, String> producer;
+  Properties properties = new Properties();
+
+  public KafkaEventPublisher(KafkaEventConfiguration kafkaConfig) {
+    super(kafkaConfig.getBatchSize(), new ArrayList<>());
+    this.kafkaEventConfiguration = kafkaConfig;
+    if (kafkaConfig.getSecurityProtocol().equals(KafkaEventConfiguration.SecurityProtocol.SSL)) {
```

Comment:
there are way too many combinations of Kafka security. SASL, SASL_SSL etc.. we should take key/value pairs as the config and pass them in as it is into kafkaConfig